bin/dup-pr

```diff
@@ -0,0 +1,273 @@
+            exit 1
+            ;;
+    esac
+done
+shift $(( $OPTIND - 1 ))
+
+prNums=$*
+[[ -n "$prNums" ]] || fatal "missing PR-NUMS argument"
+
+[[ -n "$REPO" ]] || fatal "missing -C REPO argument"
+
+# Version (can help when getting surprised by curl'ing and caching).
+echo "dup-pr version $VERSION"
+
+# Working temp dir.
+wrkDir=/var/tmp/dup-pr-$(date '+%Y%m%dT%H%M%S')
+echo "Temporary working dir: $wrkDir"
+mkdir $wrkDir
+cd $wrkDir
+
+repository=$(git -C $REPO remote get-url origin | sed -e 's/.git$//g')
+is_git=$(echo $repository | grep ^git@ || true)
+is_http=$(echo $repository | grep '^https?://' || true)
+project=""
+if [[ -n "$is_git" ]]; then
+    project=$(echo $repository | awk -F: '{print $2}')
+elif [[ -n "is_http" ]]; then
+    project=$(echo $repository | awk -F/ '{print $4 "/" $5}')
+fi
+
+if [[ -z "$project" ]]; then
+    fatal "unable to determine GitHub 'organization/repo' name for $repository"
+fi
+
+echo $project
+
+# some validation.
+if [[ ${project} != joyent/* ]]; then
+    fatal "'$project' not a joyent/NAME repo: this script doesn't support those"
+fi
+
+repoName=$(basename $project)
+
+# Get a repo clone to work with.
+echo ""
+echo "# Getting a local clone of git@github.com:$project"
+git clone git@github.com:$project $repoName
+cd $repoName
+
+for prNum in $*; do
+    echo ""
+    echo "# Duplicating https://github.com/joyent/$repoName/pull/$prNum"
+    echo ""
+
+    hub pr checkout $prNum
+
+    mkdir dup-pr-$prNum
+    chpr="dup-pr-$prNum"
+
+    current_branch=$(git rev-parse --abbrev-ref HEAD)
+    new_branch="${current_branch}-dup-pr"
+
+    # Get all of the data from the PR
+    hub api /repos/$project/pulls/$prNum > $chpr/pr.json
+
+    current_url=$(json -f $chpr/pr.json html_url)
+    current_title=$(json -f $chpr/pr.json title)
+    current_submitter=$(json -f $chpr/pr.json user.login)
+    current_state=$(json -f $chpr/pr.json state)
+    current_reviewers=$(json -f $chpr/pr.json requested_reviewers | json -a login)
+
+    # guard against closing non joyent-automation PRs and non-open PRs
+    if [[ "$current_submitter" != "joyent-automation" && -z "$ALLOW_NON_JOYENT_AUTOMATION" ]]; then
+       echo "WARNING: PR $prNum was submitted by $current_submitter not joyent-automation!"
+       echo "skipping $prNum"
+       continue
+    fi
+
+    if [[ "$current_state" != "open" ]]; then
+        echo "WARNING: Can only duplicate open PRs. PR $prNum is in state '$current_state'"
+        echo "skipping $prNum"
+        continue
+    fi
+
+    echo $current_title > $chpr/body.md
+    echo "" >> $chpr/body.md
+    echo "**This PR was migrated from $current_url**" >> $chpr/body.md
+    echo ""  >> $chpr/body.md
+
+    # No need to emit output for PRs with a 'null' body field
+    json -a -c 'this.body' -f $chpr/pr.json body >> $chpr/body.md
+
+    # Deal with comments
+    hub api /repos/$project/issues/$prNum/comments > $chpr/comments.json
+    HAS_COMMENTS=$(json -f $chpr/comments.json -ac 'this')
```

Comment:
I've never used this form, but I guess it works, so all good:

```
$ echo '[]' | json -ac 'this'
$ echo '[{}]' | json -ac 'this'
{}
```

I'd have considered:

```
$ echo '[{}]' | json length
1

$ echo '[]' | json length
0
```

and checked the number -gt 0.